packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/modules/DevLauncherInternalModule.kt

```diff
@@ -118,13 +112,8 @@ class DevLauncherInternalModule(reactContext: ReactApplicationContext?) :
   fun loadApp(url: String, promise: Promise) {
     controller.coroutineScope.launch {
       try {
-        val parsedUrl = Uri.parse(url.trim())
-        val appUrl = if (isDevLauncherUrl(parsedUrl)) {
-          requireNotNull(getAppUrlFromDevLauncherUrl(parsedUrl)) { "The provided url doesn't contain the app url." }
```

Comment:
Since this particular check has been received, could you make sure that this case (deep-link-style URL with no `url` query param) is part of the test plan and still shows a useful error message?

Reply:
@esamelson @lukmccall  I've updated the initial PR description to include some (hopefully all) of the different scenarios for handling links - can you have a look and see if there is anything missing? I will add this to the release guide as well based on feedback, and of course test these on this PR as well

Reply:
Awesome, looks comprehensive! The only thing I would suggest is adding a few malformed URLs as well, to make sure our error messages/states are correct -- maybe things like `scheme://expo-development-client` or `scheme://expo-development-client/?url=blah`.